core/src/main/java/org/infinispan/container/offheap/BoundedOffHeapDataContainer.java

```diff
@@ -223,7 +224,9 @@ private void ensureSize() {
                      addressToRemove, currentSize, maxSize);
             }
             try {
+               InternalCacheEntry<WrappedBytes, WrappedBytes> ice = offHeapEntryFactory.fromMemory(addressToRemove);
```

Comment:
I wonder if we could change this so we lazily ressurect this object. This way if no listener is installed we wouldn't have to read this object. But that would probably require changing `evictionManager` API. So maybe for ISPN 10? @tristantarrant WDYT?

Reply:
Sure.